app/Filters/OwnedIdGuard.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Filters;
+
+use CodeIgniter\Filters\FilterInterface;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\ResponseInterface;
+
+/**
+ * Class OwnedIdGuard
+ */
+class OwnedIdGuard implements FilterInterface
+{
+    /**
+     * Check if the is giving to the function arguments his own id and not the id of someone else
+     * @param RequestInterface $request
+     * @param null $arguments
+     * @return \CodeIgniter\HTTP\RedirectResponse|mixed|void
+     */
+    public function before(RequestInterface $request, $arguments = null)
+    {
+        //this will check if the id of the user is the same as the id of the user in the url
```

Comment:
```suggestion
        //this will check if the id of the user is the same as the id of the user in the url
        $idUrl = preg_split("#/#", $_SERVER['REQUEST_URI'] )[2];
```